flocx_market/tests/system/test_offer.py

```diff
@@ -0,0 +1,89 @@
+from flocx_market.db.sqlalchemy.offer_api import OfferApi
+from flocx_market.tests.base_test import BaseTest
+import json
+import datetime
+
+now = datetime.datetime.utcnow()
+
+
+class TestOffer(BaseTest):
+
+    def setUp(self):
+        super(TestOffer, self).setUp()
+        # perform auth in future
+        pass
+
+    def test_get_offer_not_found(self):
+        with self.app() as client:
+            with self.app_context():
+                res = client.get('/offers/b711b1ca-a77e-4392-a9b5-dc84c4f469ac')
+                self.assertEqual(res.status_code, 404)
+
+    def test_get_offer(self):
+        with self.app() as client:
+            with self.app_context():
+                offer = OfferApi('b9752cc0-9bed-4f1c-8917-12ade7a6fdbe',
+                                 'Alice1992',
+                                 now, 'available', 'Dell380', now,
+                                 now,
+                                 {
+                                     "new attribute XYZ": "This is just a sample list of free-form attributes used for describing a server.",
+                                     "cpu_type": "Intel Xeon", "cores": 16, "ram_gb": 512,
+                                     "storage_type": "samsung SSD", "storage_size_gb": 204},
+                                 11.5)
+                offer.save_to_db()
+                res = client.get(f'/offers/{offer.marketplace_offer_id}')
```

Comment:
There's a typo here - an extra `f`

Reply:
Looks like this happens a couple of times in this file

Reply:
that is a new f string that python 3.6  and above support: https://realpython.com/python-f-strings/

Reply:
Ah, got it - yeah, I was using the wrong version of python